android/src/main/java/com/learnium/RNDeviceInfo/RNDeviceModule.java

```diff
@@ -349,6 +349,31 @@ public String getCarrierSync() {
   @ReactMethod
   public void getCarrier(Promise p) { p.resolve(getCarrierSync()); }
 
+  @ReactMethod(isBlockingSynchronousMethod = true)
+  public List<String> getCarriersSync() {
+    List<String> carriersNames = new ArrayList<>();
+    if (telMgr != null) {
```

Comment:
The `telMgr` will always be null here because you have not actually gotten the telephony manager in this method anywhere, it's a method variable not a class instance variable